internal/service/privatelinkendpoint/resource_privatelink_endpoint.go

```
@@ -226,6 +225,9 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
 		return diag.FromErr(fmt.Errorf(ErrorPrivateLinkEndpointsSetting, "service_attachment_names", privateLinkID, err))
 	}
 
+	if privateEndpoint.GetErrorMessage() != "" {
+		return diag.FromErr(fmt.Errorf("privatelink endpoint is in a failed state: %s", privateEndpoint.GetErrorMessage()))
+	}
```

It is possible that this fix might not be touched ever. Given that only 3 fields (bar timeout) can be modified for this resource, the only failure I can conceive of which would allow for a successful POST and error in GET is an unexpected error from Atlas or the cloud provider. Further investigation would be needed to verify this. 

As it's a simple fix which adds minimal computation time, I suggest keeping the fix in the case a scenario is revealed to trigger this code to save the user from a silent fail.

---

Uncertain about raising an error here, I think a warning might be more appropriate?
What is the UX if the error is raised? Is there anything they can do other than `terraform state rm` or `terraform destroy` to get to a working state?

---

To my knowledge, there isn't an alternative for UX other than destroying the resource as the API don't have a PATCH endpoint to update this resource. I think this might make an error appropriate. WDYT @EspenAlbert ?